keps/sig-node/5018-dra-adminaccess/README.md

```diff
@@ -0,0 +1,687 @@
+
+- `k8s.io/kubernetes/pkg/registry/resource/resourceclaim`: %
+- `k8s.io/kubernetes/pkg/registry/resource/resourceclaimtemplate`: %
+
+##### Integration tests
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+-->
+
+##### e2e tests
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+
+We expect no non-infra related flakes in the last month as a GA graduation criteria.
+-->
+
+End-to-end testing depends on a working resource driver and a container runtime
+with CDI support. A
+[test driver](https://github.com/kubernetes/kubernetes/tree/master/test/e2e/dra/test-driver)
+was developed as part of the overall DRA development effort. We have extended
+this test driver to enable `DRAAdminAccess` feature gate and added tests to
+ensure `ResourceClaim` or `ResourceClaimTemplate` with `adminAccess: true`
+requests are only authorized if their namespace has the
+`kubernetes.io/dra-admin-access` label as described in this KEP.
+
+Test links:
+
+- [E2E](https://github.com/kubernetes/kubernetes/tree/master/test/e2e/dra):
+  https://testgrid.k8s.io/sig-node-dynamic-resource-allocation#ci-kind-dra
+
+Kubernetes e2e suite.[It] [sig-node] DRA [Feature:DynamicResourceAllocation]
+[FeatureGate:DynamicResourceAllocation] [Beta] cluster validate
+ResourceClaimTemplate and ResourceClaim for admin access
+[Feature:DRAAdminAccess] [FeatureGate:DRAAdminAccess] [Alpha]
+[FeatureGate:DynamicResourceAllocation] [Beta]
+
+### Graduation Criteria
+
+#### Alpha
+
+- Feature implemented behind a feature flag
+- Initial e2e tests completed and enabled
+
+#### Beta
+
+- Gather feedback
+- Additional tests are in Testgrid and linked in KEP
+
+#### GA
+
+- Allowing time for feedback
+
+### Upgrade / Downgrade Strategy
+
+The usual Kubernetes upgrade and downgrade strategy applies for in-tree
+components.
+
+### Version Skew Strategy
+
+This feature affects the kube-apiserver, kube-controller-manager, and
+kube-scheduler. The kube-apiserver must validate requests containing
+`adminAccess: true` regardless of the version of other components. The
+kube-controller-manager and kube-scheduler should ignore `adminAccess: true` if
+they are at an older version that does not support it and process the field only
+if the feature gate is detected.
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
```

Comment:
Given that DRAAdminAccess is dependent on DRAStuctured parameters, should we mention that here for disablement?

If someone disables the feature gate for DRAStructured parameters, does it make sense for this feature gate to be on?

Reply:
It doesn't do anything in that case, but let's allow such a configuration.

We used to have a sanity check where the apiserver refused to start when DRAAdminAccess was enabled and DynamicResourceAllocation was disabled. That broke clusters using `AllAlpha=true` (a reasonable setup, used by the generic "alpha" kind job) because then DRAAdminAccess is enabled while DynamicResourceAllocation is disabled (default off!).

Reply:
@ritazh One thing I am unclear on for this KEP is does it really make sense to promote StructuredParameters to GA without this KEP?

This seems to read as a good feature to have for more security conscious organizations but if StructuredParameters goes GA while this feature is in Alpha (or Beta) then we can potentially support a less than ideal scenario for a security minded organization.

Reply:
If this feature is disabled, then there is no concern about security because the functionality which needs the additional privileges (admin access) is also disabled.